pkg/ddc/base/runtime_helper.go

```diff
@@ -53,27 +53,28 @@ func init() {
 }
 
 // GetTemplateToInjectForFuse gets template for fuse injection
-func (info *RuntimeInfo) GetTemplateToInjectForFuse(pvcName string, option common.FuseSidecarInjectOption) (template *common.FuseInjectionTemplate, err error) {
+func (info *RuntimeInfo) GetTemplateToInjectForFuse(pvcName string, pvcNamespace string, option common.FuseSidecarInjectOption) (template *common.FuseInjectionTemplate, err error) {
 	if utils.IsTimeTrackerDebugEnabled() {
 		defer utils.TimeTrack(time.Now(), "RuntimeInfo.GetTemplateToInjectForFuse",
-			"pvc.name", pvcName, "pvc.namespace", info.GetNamespace())
+			"pvc.name", pvcName, "pvc.namespace", pvcNamespace)
 	}
 	// TODO: create fuse container
 	ds, err := info.getFuseDaemonset()
 	if err != nil {
 		return template, err
 	}
 
-	dataset, err := utils.GetDataset(info.client, info.name, info.namespace)
+	// Note: get the pvc corresponding dataset
+	pvcDataset, err := utils.GetDataset(info.client, pvcName, pvcNamespace)
```

Comment:
I suggest renaming `pvcDataset` to `Dataset`?

Reply:
fixed